adb.js

```diff
@@ -324,6 +325,18 @@ const ADB = {
     socket.onclose = function() {
       console.log("trackDevices onclose");
       Services.obs.notifyObservers(null, "adb-track-devices-stop", null);
+
+      // When we loose connection to the server,
+      // and the adb is still on, we most likely got our server killed
+      // by local adb. So we do try to reconnect to it.
+      setTimeout(function () { // Give some time to the new adb to start
```

Comment:
So, I suppose this works, but on Mac, it requires replugging to actually succeed.

What's worse is that we leave the non-functional device in WebIDE's devices list (see `scanner.js`), but choosing gives `PROTOCOL_ERROR`. If I replug, it will work correctly.

So, we need to clear out the device list.